calico_containers/pycalico/util.py

```diff
@@ -71,3 +75,9 @@ def get_host_ips(version=4, exclude=None):
 
     return ip_addrs
 
+def get_hostname():
+    try:
+        hostname = os.environ[HOSTNAME_ENV]
+    except KeyError:
+        hostname = socket.gethostname()
```

Comment:
I think we should fail hard if the environment variable isn't set.  Much easier to understand for the user than defaulting.  HOSTNAME is a standard environment variable, so I don't think we're putting any significant config burden on end users.